### ISSUE TITLE
csidriver register failed and kubelet will not retry

### ISSUE THREAD

**Issue by Black-max12138** [kubernetes/kubernetes#127108]
### What happened?

https://github.com/kubernetes/kubernetes/blob/95956671d8da7783a726133709b8085f56dda052/pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go#L124-L126
When Kubelet registers the CSI, if the registration of the CSI plug-in fails due to some reasons, Kubelet notifies the CSI of the registration failure but does not retry. Is this reasonable?
Whether the retry operation is performed by the csi plug-in or by the kubelet

### What did you expect to happen?

The csi plug-in registration mechanism must have a retry mechanism to ensure reliable operation.

### How can we reproduce it (as minimally and precisely as possible)?

Create some network errors when the csi plug-in is registered.

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```
1.28
</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by xing-yang** [comment]
The CSI node-driver-registrar should restart when registration fails.  Did you check that?
https://github.com/kubernetes-csi/node-driver-registrar/blob/v2.12.0/cmd/csi-node-driver-registrar/main.go#L110

/triage needs-information

**Comment by AnishShah** [comment]
sig-node CI meeting:

/remove-sig node

**Comment by Black-max12138** [comment]
> The CSI node-driver-registrar should restart when registration fails. Did you check that? https://github.com/kubernetes-csi/node-driver-registrar/blob/v2.12.0/cmd/csi-node-driver-registrar/main.go#L110
> 
> /triage needs-information

Yes, our csi plug-in also exits when registration fails. Is this the csi standard?
